app/views/search/index.html.erb

```diff
@@ -0,0 +1,44 @@
+<%= render 'search/form', text: params[:text], back_path: cookies[:back_path] %>
+
+<% if @items.present? && @items.count.positive? %>
+  <div class="items" id='items_container'>
+    <%- @items.each do |item| %>
+      <%= render 'item', item: item %>
+    <%- end %>
+  </div>
+<% else %>
+  <% unless params[:text].nil? %>
+    No results
+  <% end %>
+<% end %>
+
+<script>
+  let nextPage = 2;
+  let totalPage = <%= @total_pages %>;
+  let isCalling = false;
+
+  function callback(body) {
+    let element = document.createElement('div');
+    element.innerHTML = body;
+    document.getElementById('items_container').append(element);
+    nextPage += 1;
+    if (nextPage <= totalPage) {
+      isCalling = false;
+    }
+  }
+
+  function listen() {
+    height = document.documentElement.offsetHeight,
+		scrollingElement = document.scrollingElement || document.documentElement;
+    offset = scrollingElement.scrollTop + window.innerHeight;
+
+    if (!isCalling && offset >= height - 200) {
```

Comment:
stupid way to do load more, if there is any better js lib you wanna use, I am happy to change.